### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by celery6**
- move the coeffs polynomial data into a static var because having it inside the function was pushing like 400bytes onto the stack lol
- ifdef printf_
- add task usage otits to check if any task is using too much or too little stack (too little stack means we could reduce the stack size to reduce wasted space)
  - reduced vn task stack size now that its not using a 3000byte char array
  - add more #ifdef for icm methods
  - make main.c call error_handler if any inits fail. This should catch any init failures during last minute crunch time changes. We should decide what to do with `Error_Handler()` in production, like try to restart the board?? or just be so certain that it will never happen...?

### CODE REVIEW

File: STM32Cube/Calculations/trajectory_lib.c
Comment (celery6) [inline STM32Cube/Calculations/trajectory_lib.c:1]: the changes here are just refactoring

File: STM32Cube/Core/Src/main.c [hunk 1]
```diff
@@ -136,6 +136,42 @@ Otits_Result_t test_defaultTaskFail() {
...
+    stackWatermarks[0] = uxTaskGetStackHighWaterMark(logTaskhandle);
+    stackWatermarks[1] = uxTaskGetStackHighWaterMark(VNTaskHandle);
+    stackWatermarks[2] = uxTaskGetStackHighWaterMark(trajectoryTaskHandle);
+    stackWatermarks[3] = uxTaskGetStackHighWaterMark(stateEstTaskHandle);
+    stackWatermarks[4] = uxTaskGetStackHighWaterMark(canhandlerhandle);
+    stackWatermarks[5] = uxTaskGetStackHighWaterMark(healthChecksTaskHandle);
+    stackWatermarks[6] = uxTaskGetStackHighWaterMark(controllerHandle);
+    stackWatermarks[7] = uxTaskGetStackHighWaterMark(flightPhaseHandle);
+    stackWatermarks[8] = uxTaskGetStackHighWaterMark(oTITSHandle);
+
```
Comment (celery6) [inline STM32Cube/Core/Src/main.c:154]: someone should run OTITS to check the results of this so we dont stack overflow by accident

File: STM32Cube/Tasks/trajectory.c [hunk 2]
```diff
@@ -71,28 +71,32 @@ void trajectory_task(void* argument) {
...
+                    float vely = (altTime.alt-prev_alt)*1000.0/(altTime.time-prev_time);
+
+                    //if we see velocity drop, we know apogee is incoming regardless
+                    //to prevent the bit being prematurely set at startup due to weird numerical stuff, we only check this condition while in coast phase
+                    if(extensionAllowed() && (vely < RECOVERY_MIN_VELOCITY) ) xEventGroupSetBits(flightPhaseEventsHandle, RECOVERY_DEPLOYMENT_BIT);
+
+                    float velx = vely*tan(angles.angle.pitch);
+                    float apogee = get_max_altitude(vely,velx, altTime.alt, ext, ROCKET_BURNOUT_MASS);
+                    logInfo("traj", "%fm", apogee);
+                    xQueueOverwrite(apogeeQueue, &apogee);
```
Comment (celery6) [inline STM32Cube/Tasks/trajectory.c:85]: this change is just refactoring tabs so it doesnt look like we're missing a bracket (which is a really concerning sight)

**Review by Joe-Joe-Joe-Joe — COMMENTED** [review]
Looks good to me. I will let Armaan approve after testing this afternoon.

File: STM32Cube/Core/Src/main.c (hunk 1 above)
Comment (Armaan-Sengupta) [reply STM32Cube/Core/Src/main.c:154]: We should give OTITs more space, its currently chilling at 49 bytes remaining on its stack during runtime.

**Review by Armaan-Sengupta — DISMISSED** [review]
Increase size of OTITS stack

**Review by Joe-Joe-Joe-Joe — APPROVED** [review]
lgtm

**Pull request merged**